fvm/src/gas/price_list.rs

```diff
@@ -913,56 +929,281 @@ impl Rules for WasmGasPrices {
+            I64Load8S,
+            I64Load16S,
+            I64Load32S,
+            => fixed(self.instruction_default + self.memory_access_cost),
+
+            // Stores cost an instruction and a base fill fee.
+            F32Store, I32Store, I32Store8, I32Store16,
+            F64Store, I64Store, I64Store8, I64Store16, I64Store32,
+            TableSet,
+            => fixed(self.instruction_default + self.memory_fill_base_cost),
+
+            // Bulk memory copies & fills
+            TableInit, TableCopy => linear(
+                self.instruction_default + self.memory_access_cost,
                 self.memory_copy_per_byte_cost,
                 TABLE_ELEMENT_SIZE,
             ),
-            Operator::TableFill { .. } | Operator::TableGrow { .. } => linear_cost(
-                self.exec_instruction_cost + self.memory_fill_base_cost,
+            TableFill, TableGrow => linear(
+                self.instruction_default + self.memory_fill_base_cost,
                 self.memory_fill_per_byte_cost,
                 TABLE_ELEMENT_SIZE,
             ),
-            Operator::MemoryGrow { .. } => linear_cost(
-                self.exec_instruction_cost + self.memory_fill_base_cost,
+            MemoryGrow => linear(
+                self.instruction_default + self.memory_fill_base_cost,
                 self.memory_fill_per_byte_cost,
                 // This is the odd-one out because it operates on entire pages.
                 wasmtime_environ::WASM_PAGE_SIZE,
             ),
-            Operator::MemoryFill { .. } => linear_cost(
-                self.exec_instruction_cost + self.memory_fill_base_cost,
+            MemoryFill => linear(
+                self.instruction_default + self.memory_fill_base_cost,
                 self.memory_fill_per_byte_cost,
                 1,
             ),
-            Operator::MemoryInit { .. } | Operator::MemoryCopy { .. } => linear_cost(
-                self.exec_instruction_cost + self.memory_copy_base_cost,
+            MemoryInit, MemoryCopy => linear(
+                self.instruction_default + self.memory_access_cost,
                 self.memory_copy_per_byte_cost,
                 1,
             ),
-            // Everything else...
-            _ => Ok(InstructionCost::Fixed(
-                self.exec_instruction_cost.as_milligas() as u64,
-            )),
+
+            // Dropping is an optimization hint and probably shouldn't cost anything. But we don't
+            // use this right now anyways.
+            // TODO(M2.2) consider making this free.
+            DataDrop, ElemDrop => fixed(self.instruction_default),
+
+            // Charge one instruction for getting a table/memory size.
+            MemorySize, TableSize => fixed(self.instruction_default),
+
+            /******************/
+            /*  Unsupported   */
+            /******************/
+
+            // Exception handling.
+
+            Try, Catch, Throw, Rethrow, CatchAll, Delegate,
```

Comment:
These instructions have never been supported, we're just explicitly handling them now.